src/PointwiseFunctions/AnalyticData/ForceFree/RotatingDipole.cpp

```diff
@@ -0,0 +1,137 @@
+#include "PointwiseFunctions/InitialDataUtilities/InitialData.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace ForceFree::AnalyticData {
+
+RotatingDipole::RotatingDipole(const double vector_potential_amplitude,
+                               const double varpi0, const double delta,
+                               const double angular_velocity,
+                               const double tilt_angle,
+                               const Options::Context& context)
+    : vector_potential_amplitude_(vector_potential_amplitude),
+      varpi0_(varpi0),
+      delta_(delta),
+      angular_velocity_(angular_velocity),
+      tilt_angle_(tilt_angle) {
+  if (varpi0 < 0.0) {
+    PARSE_ERROR(context, "The length constant varpi0 ("
+                             << varpi0_ << ") cannot be negative");
+  }
+  if (delta < 0.0) {
+    PARSE_ERROR(context,
+                "The small number delta (" << delta_ << ") cannot be negative");
+  }
+  if (abs(angular_velocity) >= 1.0) {
+    PARSE_ERROR(context, "The rotation angular velocity ("
+                             << angular_velocity_
+                             << ") must be between -1.0 and 1.0");
+  }
+  if ((tilt_angle < 0.0) or (tilt_angle > M_PI)) {
+    PARSE_ERROR(context, "The rotator tilt angle ("
+                             << tilt_angle_ << ") must be between 0 and Pi");
+  }
+}
+
+RotatingDipole::RotatingDipole(CkMigrateMessage* msg) : InitialData(msg) {}
+
+std::unique_ptr<evolution::initial_data::InitialData>
+RotatingDipole::get_clone() const {
+  return std::make_unique<RotatingDipole>(*this);
+}
+
+void RotatingDipole::pup(PUP::er& p) {
+  InitialData::pup(p);
+  p | vector_potential_amplitude_;
+  p | varpi0_;
+  p | delta_;
+  p | angular_velocity_;
+  p | tilt_angle_;
+  p | background_spacetime_;
+}
+
+PUP::able::PUP_ID RotatingDipole::my_PUP_ID = 0;
+
+tuples::TaggedTuple<Tags::TildeE> RotatingDipole::variables(
+    const tnsr::I<DataVector, 3>& coords, tmpl::list<Tags::TildeE> /*meta*/) {
+  return {make_with_value<tnsr::I<DataVector, 3>>(coords, 0.0)};
+}
+
+tuples::TaggedTuple<Tags::TildeB> RotatingDipole::variables(
+    const tnsr::I<DataVector, 3>& coords,
+    tmpl::list<Tags::TildeB> /*meta*/) const {
+  auto result = make_with_value<tnsr::I<DataVector, 3>>(coords, 0.0);
+
+  const double sin_alpha = sin(tilt_angle_);
+  const double cos_alpha = cos(tilt_angle_);
+
+  // Coordinates and magnetic fields in the tilted axis
+  const auto& x = get<0>(coords);
+  const auto& y = get<1>(coords);
+  const auto& z = get<2>(coords);
+  const DataVector x_prime = cos_alpha * x - sin_alpha * z;
+  const DataVector z_prime = sin_alpha * x + cos_alpha * z;
+
+  auto tilde_b_prime = make_with_value<tnsr::I<DataVector, 3>>(coords, 0.0);
+
+  // Regularized dipole field
+  const DataVector r_squared = get(dot_product(coords, coords));
+  const DataVector sqrt_r2_plus_d2 = sqrt(r_squared + square(delta_));
```

Comment:
You only use this in expressions like `1 / pow<5>(sqrt_r2_plus_d2)`, so might as well store that instead.